Assignment 4/assignment-4-submission-folder/battle.cpp

```diff
@@ -0,0 +1,202 @@
+#include "battle.h"
+#include <iostream>
+#include <cstdlib>
+using namespace std;
+
+// Enemy Struct
+struct Enemy {
+    const char* name;
+    const char* description;
+    int damage;
+    int health;
+};
+
+// Get enemy based on its type
+Enemy* get_enemy_of_type(Enemy_Type enemyType) {
+    Enemy* enemy = new Enemy();
+    switch (enemyType) {
+        case Shrimp:
+            enemy->name = "Shrimp";
+            enemy->description = "This small agile creature attacks quick and painfully!";
+            enemy->damage = 100;
+            enemy->health = 5;
+            break;
+        case Mosquito:
+            enemy->name = "Mosquito";
+            enemy->description = "...";
+            enemy->damage = 10;
+            enemy->health = 30;
+            break;
+        case Mimic:
+            enemy->name = "Mimic";
+            enemy->description = "...";
+            enemy->damage = 10;
+            enemy->health = 40;
+            break;
+        case Dragon:
+            enemy->name = "Dragon";
+            enemy->description = "...";
+            enemy->damage = 30;
+            enemy->health = 100;
+            break;
+    }
+    return enemy;
+}
+
+// Delete an enemy
+Enemy* delete_enemy(Enemy* enemy) {
+    delete[] enemy->name;
+    delete[] enemy->description;
+    delete enemy;
+    return nullptr;
+}
+
+// Battle function
+void battle_turn(int playerIndex, Party* party, Enemy* enemy) {
```

Comment:
personally i think taking a `Player*` as an argument is much easier here (ex. not having to do memory gymnastics on line 66)